### DISCUSSION TITLE
CSC calculation for 2 impurities

### DISCUSSION THREAD

**Discussion by quantumable** [TRIQS/dft_tools#212] [General]
Dear Alex,
following your indication of our last exchange I used the script dmft here{[dmft.py.zip](https://github.com/TRIQS/dft_tools/files/8259754/dmft.py.zip)} to make the calculation dmft which besides worked by making a direct calculation but if I restore the calculation with the same file .h5 I have an error at this line: `S[icrsh].Sigma_iw = ar['DMFT_results']['Iterations']['Sigma_it'+str(iteration_offset-1)+'_'+str(icrsh)]`

even doing the csc calculation with the script here{[dmft_csc.py.zip](https://github.com/TRIQS/dft_tools/files/8259756/dmft_csc.py.zip)} i have the same error. please can you look at the script and tell me what changes to make? 

Thank you.

**Comment by the-hampel** [comment]
Hi,
could you please post the error here and describe your problem? I don’t have time to run your scripts and check for errors. You have to provide more information. 

- Alex

**Reply by quantumable** [reply]
this is the error message:


```
Starting run with 24 MPI rank(s) at : 2022-03-16 07:57:07.011049
Traceback (most recent call last):
  File "/a/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/a/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/TRIQS/install/lib/python3.8/site-packages/triqs_dft_tools/converters/plovasp/sc_dmft.py", line 257, in <module>
    main()
  File "/install/lib/python3.8/site-packages/triqs_dft_tools/converters/plovasp/sc_dmft.py", line 254, in main
    run_all(vasp_pid, dmft_mod.dmft_cycle, cfg_file, n_iter, n_iter_dft, vasp_version)
  File "/TRIQS/install/lib/python3.8/site-packages/triqs_dft_tools/converters/plovasp/sc_dmft.py", line 139, in run_all
    corr_energy, dft_dc = dmft_cycle()
  File "/free/csc.py", line 71, in dmft_cycle
    S[icrsh].Sigma_iw = ar['DMFT_results']['Iterations']['Sigma_it'+str(iteration_offset-1)+'_'+str(icrsh)]
IndexError: list index out of range
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
--------------------------------------------------------------------------
```

**Reply by the-hampel** [reply]
Hi, 
the error is pretty self-explanatory or not? The variable `icrsh` of your solver list `S[icrsh]`is out of range. Meaning that you tried to access a impurity problem `icrsh` for which no solver has been initialized. So you have to make sure that you initialize the solver object before correctly.

**Comment by quantumable** [comment]
when I do a direct dmft calculation it runs fine but if I restart the calculation using the existing .h5 file it shows me this error! yes I have initialized icrsh with  `for icrsh in range(SK.n_inequiv_shells):`
```
      for icrsh in range(SK.n_inequiv_shells):
            S[icrsh].Sigma_iw = ar['DMFT_results']['Iterations']['Sigma_it'+str(iteration_offset-1)+'_'+str(icrsh)]

```
I have attached the two scripts.

**Reply by quantumable** [reply]
hello sir,
please have you found a idea?

thank you

**Reply by the-hampel** [reply]
As I said you have to define your solver object earlier if you want to load something into it. Currently you are creating the solver in line 107 of dmft.py:
```
S.append(Solver(beta=beta, gf_struct=gf_struct))
```
but you try to load already in line 71 to write into it: 
```
S[icrsh].Sigma_iw = ar['DMFT_results']['Iterations']['Sigma_it'+str(iteration_offset-1)+'_'+str(icrsh)]
```
This cannot work. You have to create the solver object earlier to access it. Before you write something into it. 

- Alex
